### PR TITLE
Fix codesniffer errors and warnings

### PR DESCRIPTION
**Pull request by tapasweni-pathak**
* Fixed  model/ , controller/ and admin/model, admin/controller and some more.
* <del>Still need to run phpcbf to fix line ending and tab to space issues in controller/, model/.</del>
* Commits having phpcbf automatic fixes added.

### CODE REVIEW

File: admin/model/class.ArticleBackend.php [hunk 1]
```diff
@@ -1,49 +1,57 @@
...
+ * PHP Version 5.
  *
- * @author Pragya Gupta <pragya18nsit[at]gmail[dot]com>
- * @author Konstantinos Papapanagiotou <conpap[at]gmail[dot]com>
- * @license http://www.gnu.org/licenses/gpl.html
+ * @author    Pragya Gupta <pragya18nsit@gmail.com>
+ * @author    Konstantinos Papapanagiotou <conpap@gmail.com>
  * @copyright 2012 OWASP
- *
+ * @license   GNU General Public Lihttp://www.gnu.org/licenses/gpl.html
```
Comment (pchaigno) [inline admin/model/class.ArticleBackend.php]: Typo on `License` here ;)

**Comment by pchaigno** [comment]
Thanks @tapasweni-pathak!
I'll try to review this during the week-end.

Could you revert (`git revert`) commit `8be9040` (with the `esapi` modifications)? I think it's best if we don't touch this.

File: admin/model/class.ArticleBackend.php (hunk 1 above)
Comment (tapasweni-pathak) [inline admin/model/class.ArticleBackend.php]: Corrected and rebased.

**Comment by tapasweni-pathak** [comment]
@pchaigno Reverted. :)

**Comment by pchaigno** [comment]
Could you merge `Hackademic:tapasweni_scs2015` into the branch of your pull request (`tapasweni-pathak:fix_codesniffer_errors`)?

File: model/common/class.ScoringRule.php [hunk 2]
```diff
@@ -10,171 +9,214 @@
...
+     *
+     * @return Details of the scoring rules of the challenge with id or false if the
+     *         id does not exist
+     */
+    public static function getScoringRuleByChallengeId($challenge_id)
+    {
+        global $db;
+        $params = array (':challenge_id' => $challenge_id );
+        $sql = "SELECT * FROM scoring_rule WHERE challenge_id= :challenge_id LIMIT 1";
+        $result_array=self::findBySQL($sql, $params);
```
Comment (pchaigno) [inline model/common/class.ScoringRule.php]: @tapasweni-pathak Looks like you forgot to change the name of this method ;)

**Comment by tapasweni-pathak** [comment]
I will do it. 

If you know that it is fine, and you are rebasing the PR and updating it, [this](https://github.com/edx/edx-platform/wiki/How-to-Rebase-a-Pull-Request#force-push-to-update-your-pull-request) says it is fine to force push. The only option as git is rejecting otherwise. That's why I'm using it.

**Comment by pchaigno** [comment]
Sure, if you prefer to rebase, let's do that! (I usually merge because it's easier and non-destructive in case anyone else pulled the branch but a rebase is cleaner for the commit history ;).

File: admin/controller/class.OptionsController.php [hunk 3]
```diff
@@ -1,237 +1,270 @@
...
-  /**
-   * The main controller of the view that controls if data has been submitted or if it's
-   * a regular page view. Then builds the view and generates it.
-   */
-  public function go() {
-    if(isset($_POST['submit'])) {
-      $this->handleFormData();
-    } else {
-      $this->setDefaultTemplateVars();
+        $this->_buildView($this->_active_plugins, $this->_active_user_theme);
```
Comment (pchaigno) [inline admin/controller/class.OptionsController.php]: You also have to change the name of these variables (here `_active_plugins`) in the associated `.tpl` files (here `admin/view/options.tpl`).

File: admin/controller/class.AddChallengeController.php [hunk 4]
```diff
@@ -1,260 +1,287 @@
...
+        } else {
+            $this->addErrorMessage("There was a problem. Please try again!");
+            return false;
+        }
+    }
+
+    public function go()
+    {
+        $this->setViewTemplate('addchallenge.tpl');
+        $this->cache_values();
```
Comment (pchaigno) [inline admin/controller/class.AddChallengeController.php]: It's `cacheValues` instead of `cache_values` ;)

**Comment by tapasweni-pathak** [comment]
For a PR :
Yes for the clean commit history I'm doing it. So that it will be easy to go through the change log later if someone want to.

You have my first PR merged in Hackademic:tapasweni_scs2015. But my task specific branch is based on next-dev, and that does not have the first PR merged. So, to avoid conflicts when you will merge it to Hackademic:tapasweni_scs2015, I will merge it on my side.

File: controller/class.ProgressReportController.php [hunk 5]
```diff
@@ -1,136 +1,145 @@
...
+            }
+
+            $data = array();
+            $class_ids = array();
+            $class_scores = array();
+            $classes_of_user = ClassMemberships::getMembershipsOfUserObjects($user->id);
+            //var_dump(UserChallenges::getChallengesOfUser($user->id));
+            foreach ($classes_of_user as $class) {
+                $progress = ChallengeAttempts::getUserProgress($user->id, $class->id);
+                $user_scores = UserScore::get_scores_for_user_class($user->id, $class->id);
```
Comment (pchaigno) [inline controller/class.ProgressReportController.php]: Looks like this is the name of an old method.

File: admin/controller/class.AddChallengeController.php [hunk 6]
```diff
@@ -1,260 +1,287 @@
...
-      return $a;
-    } else {
-      $this->addErrorMessage("There was a problem. Please try again!");
-      return false;
-	  }
-  }
-
-  public function go() {
-    $this->setViewTemplate('addchallenge.tpl');
-    $this->cache_values();
```
Comment (tapasweni-pathak) [inline admin/controller/class.AddChallengeController.php]: It is cache_values() only in the original file. Should I change it, as it seems wrong?

**Comment by pchaigno** [comment]
> You have my first PR merged in Hackademic:tapasweni_scs2015. But my task specific branch is based on next-dev, and that does not have the first PR merged. So, to avoid conflicts when you will merge it to Hackademic:tapasweni_scs2015, I will merge it on my side. 

:+1: 
Thanks!

File: admin/controller/class.OptionsController.php (hunk 3 above)
Comment (tapasweni-pathak) [inline admin/controller/class.OptionsController.php]: But it is a private variable, still it can be used in some other file?

If yes, then I have to do this for all such variables. A lot private variable names are changed now. They have a underscore now. For all of them I need to search them and correct them, then. Is it?

File: controller/class.ProgressReportController.php (hunk 5 above)
Comment (tapasweni-pathak) [inline controller/class.ProgressReportController.php]: I greped while I was doing this change. I don't know how I missed this one. Sorry. Corrected it.

File: admin/controller/class.AddChallengeController.php (hunk 6 above)
Comment (pchaigno) [inline admin/controller/class.AddChallengeController.php]: You changed the name of the method to be standard compliant (at the end of the file).

Comment (tapasweni-pathak) [inline admin/controller/class.AddChallengeController.php]: woops. Corrected it. Thanks a lot.

File: admin/controller/class.OptionsController.php (hunk 3 above)
Comment (pchaigno) [inline admin/controller/class.OptionsController.php]: > But it is a private variable, still it can be used in some other file?

Yes, because [this variable is added to the view](https://github.com/pchaigno/hackademic/blob/next-dev/admin/controller/class.OptionsController.php#L121-L125).

> For all of them I need to search them and correct them, then. Is it?

I think that searching for variables without an underscore in the `.tpl` files should be enough, no? (using some kind of regexp like `\$[^_]`)

File: controller/class.ProgressReportController.php (hunk 5 above)
Comment (pchaigno) [inline controller/class.ProgressReportController.php]: No problem, that's what the review is for ;)

File: admin/controller/class.OptionsController.php (hunk 3 above)
Comment (tapasweni-pathak) [inline admin/controller/class.OptionsController.php]: > I think that searching for variables without an underscore in the `.tpl` files should be 
> enough, no? (using some kind of regexp like `\$[^_]`)

I did not get it, are you suggesting to search for variables without an underscore and then add one in all of them? But some are private and only those are changed. I need to look what all private variable names, I have changed and then change them in the view, isn't it? or am I wrong?

One more thing I should search only in view/ folder right?

File: admin/controller/class.OptionsController.php [hunk 7]
```diff
@@ -1,237 +1,270 @@
...
+    /**
+     * Sets the view vars and template.
+     * @return Nothing.
+     */
+    private function _buildView()
+    {
+        $this->setViewTemplate('options.tpl');
+        $this->addToView('plugins', $this->_getPlugins());
+        $this->addToView('user_themes', $this->_getUserThemes());
+        $this->addToView('_active_plugins', $this->_active_plugins);
```
Comment (pchaigno) [inline admin/controller/class.OptionsController.php]: So, the error above is happening because you changed the name of the variable in the view here (the first argument). Maybe we could consider that this is out of the scope of the PSR.
I think it would make sense since it is not a variable here and in the context of the template it's more some kind of local variable. The template in then compiled into a PHP file but that's clearly out of the scope of the PSR (since it's generated).
What do you think?

Comment (tapasweni-pathak) [inline admin/controller/class.OptionsController.php]: @pchaigno  Yes if it is generated PHP file, I'm fine with leaving it. Should I undo these changes then? What all need to be undoed?

Comment (pchaigno) [inline admin/controller/class.OptionsController.php]: You only need to remove the underscores at the beginning of the variable names for the views.
So,
```php
$this->addToView('_active_plugins', $this->_active_plugins);
```
should become:
```php
$this->addToView('active_plugins', $this->_active_plugins);
```

Comment (tapasweni-pathak) [inline admin/controller/class.OptionsController.php]: @pchaigno Done. I greped 

```
grep -rl "addToView('_" *
```

Only one came out *admin/controller/class.OptionsController.php*.

**Comment by tapasweni-pathak** [comment]
Done with merging too. :smile: I think now it is ready to get merged. Let me know if some other fixes are required.

**Comment by pchaigno** [comment]
> Done with merging too.

I still can't merge (see below). You will need to merge `tapasweni_scs2015` into this branch ;)

Have you tried to install it to see if there are any errors?

**Comment by tapasweni-pathak** [comment]
> I still can't merge (see below). You will need to merge tapasweni_scs2015 into this branch ;)

Woops. I thought my next-dev and hackademic tapasweni_scs2015 is same, so merged that. Done with fixing the merging issues now.

> Have you tried to install it to see if there are any errors?

What all do you do to test the changes? [This](https://github.com/Hackademic/hackademic/blob/next-dev/CONTRIBUTING.md#running-tests) also?

**Comment by pchaigno** [comment]
> Done with fixing the merging issues now.

:+1:

> What all do you do to test the changes? This also?

That could be a good idea but currently I just go through the website to see if there are any error messages.

**Comment by tapasweni-pathak** [comment]
> That could be a good idea but currently I just go through the website to see if there are any error messages.

Go through the website, as in, any specific things you test. Do you uninstall and reinstall it or do you simply run it and test?

File: admin/controller/class.OptionsController.php [hunk 8]
```diff
@@ -1,237 +1,270 @@
...
+     */
+    private function _handleFormData()
+    {
+        // Handle plugins
+        foreach ($_POST as $key => $value) {
+            if (substr($key, 0, 7) == 'plugin_') {
+                array_push($this->_active_plugins, $value);
+            }
+        }
+        $this->_notifyEnabledPlugins(array_diff($this->_active_plugins, Options::getOption('_active_plugins')->value));
```
Comment (pchaigno) [inline admin/controller/class.OptionsController.php]: `Options` saves parameters to the database. I think you should remove the underscores here as well:
`Options::getOption('active_plugins')` instead of `Options::getOption('_active_plugins')`.

**Comment by pchaigno** [comment]
I reinstall it because there could be errors in the installation process too ;)

File: admin/controller/class.OptionsController.php [hunk 9]
```diff
@@ -1,237 +1,270 @@
...
+        $this->_notifyDisabledPlugins(array_diff(Options::getOption('_active_plugins')->value, $this->_active_plugins));
+
```
Comment (tapasweni-pathak) [inline admin/controller/class.OptionsController.php]: Here too right?

File: admin/controller/class.OptionsController.php [hunk 10]
```diff
@@ -1,237 +1,270 @@
...
-    // Handle theme
-    $new_user_theme_path =  $_POST['active_user_theme'];
-    if($this->active_user_theme != $new_user_theme_path) {
-      $this->smarty->clear_all_cache();
-      Plugin::do_action_ref_array('enable_user_theme', array($new_user_theme_path));
-      Plugin::do_action_ref_array('disable_user_theme', array($this->active_user_theme));
-    }
-    $this->active_user_theme = $new_user_theme_path;
+        // Update options in database
+        Options::updateOption('_active_plugins', $this->i_active_plugins);
```
Comment (tapasweni-pathak) [inline admin/controller/class.OptionsController.php]: Here?

Comment (pchaigno) [inline admin/controller/class.OptionsController.php]: Yes, everywhere it is used as a key for `Options`.

Comment (tapasweni-pathak) [inline admin/controller/class.OptionsController.php]: Done.

File: admin/controller/class.AddUserController.php [hunk 11]
```diff
@@ -1,112 +1,127 @@
...
+                } elseif (!Utils::validateEmail($email)) {
+                    $this->addErrorMessage("Please enter a valid email id");
+                } else {
+                    $subject="Hackademic new account";
+                    $message="Hackademic account created succesfully";
+                    //Mailer::mail($email,$subject,$message);
+                    $joined=date("Y-m-d H-i-s");
+                    $result = User::addUser(
+                        $username, $full_name, $email, $password,
+                        $joined, $activated, $typ
```
Comment (pchaigno) [inline admin/controller/class.AddUserController.php]: Small typo here on `$type` ;)

Comment (tapasweni-pathak) [inline admin/controller/class.AddUserController.php]: oops. Done. :+1:

File: model/common/class.UserScore.php [hunk 12]
```diff
@@ -10,198 +9,250 @@
...
+  *
+     * @param SQL   $sql    The SQL query.
+     * @param Param $params The parameters for the query.
+     *
+     * @return An array of UserScore objects.
+     */
+    private static function findBySQL($sql, $params = null)
+    {
+        global $db;
+        $statement_handle = $db->read($sql, $params, $_action_type);
```
Comment (pchaigno) [inline model/common/class.UserScore.php]: There is a missing `self::` here:
`self::$_action_type` instead of `$_action_type`.
Looks like this error was here before your changes though :))

File: controller/class.ChallengeMonitorController.php [hunk 13]
```diff
@@ -1,418 +1,449 @@
...
+    define('UA_PENALTY_ID', "banned_user_agent_penalty");
+    define('MULT_SOL_BONUS_ID', "multiple_solution_bonus");
+    define('TOTAL_ATTEMPT_PENALTY_ID', "total_attempt_penalty");
+    define('FTS_PENALTY_ID', "first_try_penalty");
+
+    define("CHALLENGE_INIT", 2);
+    define("CHALLENGE_SUCCESS", 1);
+    define("CHALLENGE_FAILURE", 0);
+}
+class ChallengeMonitorController
```
Comment (pchaigno) [inline controller/class.ChallengeMonitorController.php]: You removed the method `go` that was in this class before. It's called in `ChallengeValidatorController`.
I think you should leave it there, even if it's empty.

File: model/common/class.UserScore.php (hunk 12 above)
Comment (tapasweni-pathak) [inline model/common/class.UserScore.php]: Done. :)

File: controller/class.ChallengeMonitorController.php (hunk 13 above)
Comment (tapasweni-pathak) [inline controller/class.ChallengeMonitorController.php]: Done.

File: controller/class.ChallengeMonitorController.php [hunk 14]
```diff
@@ -1,418 +1,452 @@
...
+         * rest of the values
+         */
+        if ($_SESSION['token'] == $token && $token != null) {
+            
+            // User changed challenge
+            if (($pkg_name != $_SESSION['pkg_name'] && $pkg_name != null)
+                || ($_SESSION['chid'] != $chid && $chid != null)
+            ) {
+                if (!$pair) {
+                    _invalidChallenge();
```
Comment (pchaigno) [inline controller/class.ChallengeMonitorController.php]: `$this->` is missing here ;)

File: model/common/class.User.php [hunk 15]
```diff
@@ -1,321 +1,341 @@
...
+
+    public static function addToken($username, $token)
+    {
+        global $db;
+        $sql = "UPDATE users SET token=:token WHERE username = :username";
+        $params = array(
+         ':token' => $token,
+        ':username' => $username
+        );
+        $query = $db->update($sql, $params);
```
Comment (pchaigno) [inline model/common/class.User.php]: It should be `$query = $db->update($sql, $params, self::$_action_type);` instead of ` $query = $db->update($sql, $params);` here.
It was missing before.

File: controller/class.RegisterUserController.php [hunk 16]
```diff
@@ -1,108 +1,106 @@
...
+                    $message="Hackademic account created succesfully";
+                    //Mailer::mail($email,$subject,$message);
+                    $joined=date("Y-m-d H-i-s");
+                    $result = User::addUser($username, $full_name, $email, $password, $joined);
+                    $usr = User::findByUserName($username);
+                    $res2 = ClassMemberships::addMembership($usr->id, GLOBAL_CLASS_ID);
+                    $this->addSuccessMessage("You have been registered succesfully");
+                }
+            }
+        }
```
Comment (pchaigno) [inline controller/class.RegisterUserController.php]: We lost a else block here ;)
```php
else {
    $this->addToView('cached', $this);
}
```

Comment (tapasweni-pathak) [inline controller/class.RegisterUserController.php]: Fixed all.

**Comment by pchaigno** [comment]
Were you able to run some tests?

**Comment by tapasweni-pathak** [comment]
> Were you able to run some tests?

Nope. When I'm trying to download selenium-server-standalone-2.45.0.jar, [this](http://i.imgur.com/sbQ7J3Q.png) is coming up. :(

**Comment by pchaigno** [comment]
You can download the last version here: http://www.seleniumhq.org/download/

**Comment by tapasweni-pathak** [comment]
There is something as *Download version 2.47.1* in this page, there only I'm clicking to download it.

**Comment by pchaigno** [comment]
Try with `wget` or an other browser maybe. It works fine for me :/

**Comment by tapasweni-pathak** [comment]
Not working on [Firefox](http://i.imgur.com/UzII7Rl.png), [Chromium](http://i.imgur.com/LEU7vBb.png), and Opera Mini too. Can it be send over an email or in some other way?

wget says forbidden 403.

**Comment by northdpole** [comment]
the selenium jar is laready in https://github.com/Hackademic/hackademic/tree/next-dev/hackademic_devtests (it's not the latest one but it should work, I can upload the latest one if you like)

**Comment by tapasweni-pathak** [comment]
I checked reinstalling it, and going through the UI, that is fine. What else I can do?

**Comment by northdpole** [comment]
the pr seems ok, since @pchaigno is the mentor for this project it's better for him to give the final green light. 
Please submit the next one regardless

**Comment by pchaigno** [comment]
I'll do one last check of this PR this week-end or on Wednesday and I'll merge!

> Please submit the next one regardless

:+1:
